### PR TITLE
CB-600 Integration test for templating, for Cloudera Manager template

### PR DESCRIPTION
**Pull request by bbihari**

### CODE REVIEW

File: integration-test/src/main/java/com/sequenceiq/it/cloudbreak/newway/testcase/mock/ClouderaManagerShowClusterDefinitionTest.java [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+import com.sequenceiq.it.cloudbreak.newway.context.MockedTestContext;
+import com.sequenceiq.it.cloudbreak.newway.entity.AmbariEntity;
+import com.sequenceiq.it.cloudbreak.newway.entity.ClusterEntity;
+import com.sequenceiq.it.cloudbreak.newway.entity.clusterdefinition.ClusterDefinitionTestDto;
+import com.sequenceiq.it.cloudbreak.newway.entity.stack.StackTestDto;
+import com.sequenceiq.it.cloudbreak.newway.util.ShowClusterDefinitionUtil;
+
+public class ClouderaManagerShowClusterDefinitionTest extends AbstractClouderaManagerTest {
+
+    @BeforeMethod
```
Comment (topolyai5) [inline integration-test/src/main/java/com/sequenceiq/it/cloudbreak/newway/testcase/mock/ClouderaManagerShowClusterDefinitionTest.java]: You should add AfterMethod, like this: 
```
@AfterMethod(alwaysRun = true)
 public void tear(Object[] data) {
        MockedTestContext testContext = (MockedTestContext) data[0];
        testContext.cleanupTestContextEntity();
    }
```

File: integration-test/src/main/java/com/sequenceiq/it/cloudbreak/newway/util/ShowClusterDefinitionUtil.java [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+    private static void validateGeneratedClusterDefinition(String extendedClusterDefinitionText) {
+        if (Strings.isNullOrEmpty(extendedClusterDefinitionText)) {
+            throw new TestFailException("Generated Cluster Definition does not exist");
+        } else if (!isJSONValid(extendedClusterDefinitionText)) {
+            throw new TestFailException("Generated Cluster Definition is not a valid json");
+        }
+    }
+
+    private static boolean isJSONValid(String jsonInString) {
```
Comment (lnardai) [inline integration-test/src/main/java/com/sequenceiq/it/cloudbreak/newway/util/ShowClusterDefinitionUtil.java]: We have this exact same validator in this class.
JsonUtil.isValid()

File: integration-test/src/main/java/com/sequenceiq/it/cloudbreak/newway/util/ShowClusterDefinitionUtil.java [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+        return stackTestDto;
+    }
+
+    public static StackTestDto checkGeneratedClusterDefinition(TestContext testContext, StackTestDto stackTestDto, CloudbreakClient cloudbreakClient) {
+        String extendedClusterDefinitionText = stackTestDto.getResponse().getCluster().getAmbari().getExtendedClusterDefinitionText();
+        validateGeneratedClusterDefinition(extendedClusterDefinitionText);
+        return stackTestDto;
+    }
+
+    private static void validateGeneratedClusterDefinition(String extendedClusterDefinitionText) {
```
Comment (lnardai) [inline integration-test/src/main/java/com/sequenceiq/it/cloudbreak/newway/util/ShowClusterDefinitionUtil.java]: I would create a stronger assert here to test at least 1 field of the Json that is generated.

File: integration-test/src/main/java/com/sequenceiq/it/cloudbreak/newway/testcase/mock/ClouderaManagerShowClusterDefinitionTest.java (hunk 1 above)
Comment (bbihari) [reply integration-test/src/main/java/com/sequenceiq/it/cloudbreak/newway/testcase/mock/ClouderaManagerShowClusterDefinitionTest.java]: ok

**Pull request merged**